packages/react-hooks/src/hooks/delayed-callback.ts

```diff
@@ -0,0 +1,25 @@
+import {useEffect, useState} from 'react';
+
+export function useDelayedCallback(callback: () => void, delay: number) {
+  const [timeoutId, setTimeoutId] = useState<ReturnType<typeof setTimeout>>();
```

Comment:
Based on the list of dependencies of your `useEffect`, I guess this could be changed for a `useRef` instead. You don't really need to re-render when the timeout is set I believe.